.ci/Jenkinsfile

```diff
@@ -338,50 +337,53 @@ def generateFunctionalTestStep(Map args = [:]){
-      deleteDir()
-      unstash 'source'
-      withGoEnv(version: "${GO_VERSION}"){
-        try {
-          if(isInstalled(tool: 'docker', flag: '--version')) {
-            dockerLogin(secret: "${DOCKER_ELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
-          }
-          filebeat(image: 'docker.elastic.co/beats/filebeat:7.13.0-SNAPSHOT', output: "docker_logs_${goArch}_${suite}_${name}.log", workdir: "${env.WORKSPACE}") {
-              dir("${BASE_DIR}") {
-                  def envContext = []
-                  if (platform.contains("windows")) {
-                       envContext = [
-                         "KIBANA_URL=https://${linuxIp}:5601",
-                         "ELASTICSEARCH_URL=https://${linuxIp}:9200",
-                         "FLEET_URL=https://${linuxIp}:8220",
-                         "PROVIDER=remote",
-                         "GOARCH=${goArch}",
-                         "ELASTIC_APM_GLOBAL_LABELS=build_pr=${isPR()},build_id=${env.BUILD_ID},go_arch=${goArch},beat_version=${env.BEAT_VERSION},stack_version=${env.STACK_VERSION}"
-                       ]
-                       withEnv(envContext) {
-                           withSecretVault(secret: "${VAULT_INSTRUMENTATION_SECRET}", user_key: "apmServerToken", user_var_name: "APM_SECRET_TOKEN", pass_key: "apmServerUrl", pass_var_name: "APM_SERVER_URL"){
-                               cmd script: "cd e2e\\_suites\\${suite} && go test -timeout 60m -v --godog.tags=${tags}", label: "Run functional tests for ${platform}:${suite}:${tags}"
-                           }
-                       }
-                   } else {
-                       envContext = ["GOARCH=${goArch}", "ELASTIC_APM_GLOBAL_LABELS=build_pr=${isPR()},build_id=${env.BUILD_ID},go_arch=${goArch},beat_version=${env.BEAT_VERSION},stack_version=${env.STACK_VERSION}"]
-                       withEnv(envContext) {
-                           withSecretVault(secret: "${VAULT_INSTRUMENTATION_SECRET}", user_key: "apmServerToken", user_var_name: "APM_SECRET_TOKEN", pass_key: "apmServerUrl", pass_var_name: "APM_SERVER_URL"){
-                               sh script: """.ci/scripts/functional-test.sh "${suite}" "${tags}" "${STACK_VERSION}" "${BEAT_VERSION}" """, label: "Run functional tests for ${platform}:${suite}:${tags}"
-                           }
-                       }
-                   }
-              }
-          }
-        } finally {
-          junit(allowEmptyResults: true, keepLongStdio: true, testResults: "${BASE_DIR}/outputs/TEST-*.xml")
-          archiveArtifacts allowEmptyArchive: true, artifacts: "${BASE_DIR}/outputs/TEST-*.xml"
-          tearDown(labels: workerLabels)
+    // Setup environment for platform
+    def envContext = []
+    if (platform.contains("windows")) {
+        envContext = [
+            "KIBANA_URL=https://${linuxIp}:5601",
+            "ELASTICSEARCH_URL=https://${linuxIp}:9200",
+            "FLEET_URL=https://${linuxIp}:8220",
```

Comment:
I've just realised that we are not exposing fleet-server port. Will create a PR to do so now!